index/scorch/segment/zap/chunk.go

```diff
@@ -0,0 +1,40 @@
+package zap
+
+import (
+	"fmt"
+)
+
+// LegacyChunkMode was the original chunk mode (always chunk size 1024)
+// this mode is still used for chunking doc values.
+var LegacyChunkMode uint32 = 1024
+
+// DefaultChunkMode is the most recent improvement to chunking and should
+// be used by default.
+var DefaultChunkMode uint32 = 1025
+
+func getChunkSize(chunkMode uint32, cardinality uint64, maxDocs uint64) (uint64, error) {
+	switch {
+	// any chunkMode <= 1024 will always chunk with chunkSize=chunkMode
+	case chunkMode <= 1024:
+		// legacy chunk size
+		return uint64(chunkMode), nil
+
+	case chunkMode == 1025:
+		// attempt at simple improvement
+		// theory - the point of chunking is to put a bound on the maximum number of
+		// calls to Next() needed to find a random document.  ie, you should be able
+		// to do one jump to the correct chunk, and then walk through at most
+		// chunk-size items
+		// previously 1024 was chosen as the chunk size, but this is particularly
+		// wasteful for low cardinality terms.  the observation is that if there
+		// are less than 1024 items, why not put them all in one chunk,
+		// this way you'll still achieve the same goal of visiting at most
+		// chunk-size items.
+		// no attempt is made to tweak any other case
+		if cardinality <= 1024 {
```

Comment:
In early trials, one thought i had here is that - though the default CF was 1024, we ever know/measured - whether there were really 1024 entries are going inside a chunk.  So there could be perf regressions resulting from a more number of Next() calls inside a chunk with these 1024 values getting coalesced into a single chunk now.  (more document values going inside a chunk than early days)
Didn't had a better cheap fix for that, so in my toy trials - I had tried 256/512 as the actual cardinality limit for coalescing to a single chunk, as for that use case terms involved were of high uniqueness.

Reply:
Agreed, it's always been arbitrary, but sticking 1024 seemed more consistent with the original design (no one should have to call Next more than 1024 times).  But you're right there was never any reason to think that was a good choice in the first place.

Also, this current version only attempts to help with the sparsest of field-terms.  If there are 1025 docs, you're back to lots of chunks again.  And if you have 100 million docs, even 1025 is sparse...

My thought is that this helps address the immediate problem, but we can define new chunk modes in the future that do better.